compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -2813,7 +2780,7 @@ private void generateArrayAccess(BLangIndexBasedAccess astArrayAccessExpr) {
                                               astArrayAccessExpr.isLValue && !astArrayAccessExpr.leafNode));
         this.env.targetOperand = tempVarRef;
 
-        this.varAssignment = variableStore;
```

Comment:
Shall we use `this.varAssignment` directly in the `if` condition in line number 2768? Because the `variableStore` local var is unnecessary after this change.

Reply:
We change the varAssignment value initially. 
boolean variableStore = this.varAssignment;
        this.varAssignment = false;

So as per the logic we use previous this.varAssignment for if check. So we cannot directly use this.varAssignment in if condition.